### PR TITLE
Starting to document architecture and contribution instructions

### PR DESCRIPTION
**Pull request by etsauer**
#### What does this PR do?
Added docs for provisioning architecture and started a guide for contributing a provisioner

#### How should this be manually tested?
read the docs and provide feedback

#### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.

#### Who would you like to review this?
cc: @redhat-cop/casl

### CODE REVIEW

**Comment by etsauer** [comment]
@oybed @raffaelespazzoli @makentenza updated Architecture doc. Would like some feedback on this so we can move closer to merging it.

File: docs/PROVISIONING_ARCH.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- import_playbook: pre-install.yml <2>
+
+# Install - this runs openshift-ansible
+- import_playbook: install.yml <3>
+
+# Post Install
+- import_playbook: post-install.yml <4>
+```
+
+1. _Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure requires (network, DNS, load balancers, etc.). In general this phase
```
Comment (oybed) [inline docs/PROVISIONING_ARCH.md]: typo? `requires` --> `required`?

File: docs/PROVISIONING_ARCH.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed, SSH keys (with root or sudo) pushed, and `subscription-manager` installed.
```
Comment (oybed) [inline docs/PROVISIONING_ARCH.md]: the `subscription-manager` installed step I believe is a Google Cloud only, and not sure I'd call it out here.

File: docs/PROVISIONING_ARCH.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+2. _Pre-Install_ handles all of the host prep you'd expect to happen when preparing to install OpenShift. This phase includes:
+  - Subscribing the hosts via `subscription-manager` (supports individual account or Satellite subscription using _activation keys_)
+  - Installing/configuring the container runtime, including container storage devices
+  - Installing prerequisite packages
```
Comment (oybed) [inline docs/PROVISIONING_ARCH.md]: Maybe include:
```
- update (and reboot) hosts
```

File: docs/PROVISIONING_ARCH.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - Syncing additional administrator SSH keys (typically a cloud provider only provisions a single key)
+  - Creating a set of demo users (when using the `HTPassswordIdentityProvider`)
+  - Creation of any sort of OpenShift resources using the [openshift-applier](https://github.com/redhat-cop/openshift-applier.git) automation framework.
+
+Some principles we try to adhere to with all of these phases include the following.
+
+- **Modularity**. Each phase should be runnable on its own, using the exact same inventory
+- **Idempotentcy**. All phases should be safe to re-run at any time without impact.
+- **Commonality**. This is the idea that we should essentially treat all Infrastructure providers the same, allowing us to treat different providers as _pluggable_ to a [common provisioning framework](#pluggable-provisioners).
+
```
Comment (oybed) [inline docs/PROVISIONING_ARCH.md]: Should we include a note about `de-provisioning` / `deleting a cluster` ?

File: docs/PROVISIONING_ARCH.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+## OpenShift Ansible integration
+
+CASL-Ansible is intended to be a compliment (and in some cases an upstream) to the [openshift-ansible](https://github.com/openshift/openshift-ansible.git) repo. As such, we intend for the CASL tools to support all of the features and capabilities of the openshift-ansible cluster deployment playbook. The aim is to use CASL to provision and prepare infrastructure and then hand off to the [deploy_cluster](https://github.com/openshift/openshift-ansible/blob/master/playbooks/deploy_cluster.yml) playbook for the installation of OpenShift.
```
Comment (oybed) [inline docs/PROVISIONING_ARCH.md]: Maybe start this with "CASL-Ansible **provisioning** is intended ...." as CASL includes a lot more than just provisioning now.

**Pull request merged**